### ISSUE TITLE
FR: Add HTTP/3 support for nginx

### ISSUE THREAD

**Issue by patschi** [froxlor/Froxlor#1144]
**Is your feature request related to a problem? Please describe.**
The nginx 1.25.0 mainline version introduced experimental HTTP/3 support:
https://mailman.nginx.org/pipermail/nginx-announce/2023/KEVVKRTAXBWBVXA3DIF5SUKRJWZOEMTM.html

**Describe the solution you'd like**
Implement a setting to be able to enable HTTP/3 for nginx in Froxlor.

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

**Comment by patschi** [comment]
Documentation:
https://nginx.org/en/docs/http/ngx_http_core_module.html#quic
https://nginx.org/en/docs/quic.html

There are several new options available. To enable it, the config can look like:
```
server {
        listen IP:443 ssl http2;
        listen IP:443 quic;

        quic_retry on;
        ssl_early_data on;
        [...]
}
```

**Comment by d00p** [comment]
even debian 12 only has 1.22.1 in its repositories + its experimental. Hence very low priority

**Comment by hardfalcon** [comment]
FYI: nginx 1.25.1 changed the syntax for how SSL/TLS and HTTP/2 are configured:
https://nginx.org/en/CHANGES-1.26
> Changes with nginx 1.25.1                                        13 Jun 2023
>     *) Feature: the "http2" directive, which enables HTTP/2 on a per-server
>        basis; the "http2" parameter of the "listen" directive is now
>        deprecated.
>     *) Change: HTTP/2 server push support has been removed.
>     *) Change: the deprecated "ssl" directive is not supported anymore.

Would it be an option to have Froxlor support the stable nginx packages from https://nginx.org/en/linux_packages.html? The nginx devs cut a new stable version branch once per year in April or May, and the more bleeding edge "nginx mainline" branch could be used to add support for upcoming incompatible changes in nginx early on in Froxlor. The nginx change logs for both the stable branches and the mainline branch can be found [here](https://nginx.org/en/download.html)

**Comment by d00p** [comment]
We already implemented the `http2 on` for nginx >= 1.25.1, see https://github.com/froxlor/Froxlor/commit/2629718b229fc7ef7518d38d32be5a55c5224086
